internal/engine/build_and_deployer_test.go

```diff
@@ -133,25 +133,33 @@ func TestIncrementalBuildWaitsForPostProcess(t *testing.T) {
 	f := newBDFixture(t, k8s.EnvGKE)
 	defer f.TearDown()
 
-	f.k8s.SetPollForPodWithImageDelay(time.Second)
-	go f.bd.PostProcessBuild(f.ctx, alreadyBuilt) // will take 1s
-	time.Sleep(time.Millisecond * 100)            // let the PostProcessBuild call actually start
+	f.k8s.SetPollForPodWithImageDelay(time.Second * 1)
 
-	_, err := f.bd.BuildAndDeploy(f.ctx, SanchoManifest, NewBuildState(alreadyBuilt))
+	res, err := f.bd.BuildAndDeploy(f.ctx, SanchoManifest, BuildStateClean)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// Rather than falling back to image build b/c of lack of deploy info, we should
-	// wait for the concurrent call to PostProcessBuild to finish and do a container build.
-	if f.docker.BuildCount != 0 {
-		t.Errorf("Expected no docker build, actual: %d", f.docker.BuildCount)
+	// HACK: if we fail to container build here, the test has already failed. However,
```

Comment:
???

i'm confused by the statement "this test setup doesn't support incremental image build"

TestIncrementalBuildTwiceDeadPod should test this?

Reply:
AAAAHHHH i figured it out. what it doesn’t support is two image builds in a row (Test...DeadPod is some container builds and then a single image build) since the return val for `fake_client.ImageBuild` is a reader. so we read til the end in the first image build, and then there’s nothing to read in the second. Will fix in a separate PR